File: bridge/protocol.py
"""IPC op 常量 — 与 core/internal/protocol/ops.go、core/CONTRACT.md 同步。"""
from __future__ import annotations

DEFAULT_PIPE = r"\\.\pipe\liveaio-core"
DEFAULT_TCP = ("127.0.0.1", 19877)
CORE_VERSION = "0.1.0"

# 握手 / 生命周期
OP_READY = "ready"
OP_PING = "ping"
OP_PONG = "pong"
OP_SHUTDOWN = "shutdown"
OP_ERROR = "error"

# 会话 / 帧
OP_CONNECT = "connect"
OP_DISCONNECT = "disconnect"
OP_FRAME_PUSH = "frame.push"
OP_MESSAGE_INGEST = "message.ingest"
OP_STATUS = "status"
OP_MESSAGE = "message"

# 配置（Python 写文件；此处只推快照）
OP_CONFIG_SET = "config.set"
OP_CONFIG_GET = "config.get"

# 工具业务事件 Core → UI
OP_TICK = "tick"
OP_LEDGER = "ledger"
OP_DANMU_SHOW = "danmu.show"
OP_MEMO_ITEM = "memo.item"

# 工具命令 UI → Core
OP_TOOL_OVERTIME_SET = "tool.overtime.set"
OP_TOOL_OVERTIME_CMD = "tool.overtime.cmd"
OP_TOOL_OVERTIME_SIM = "tool.overtime.sim_gift"
OP_TOOL_DANMU_SET = "tool.danmu.set"
OP_TOOL_MEMO_SET = "tool.memo.set"

File: core/protocol.go
package core

// Protocol constants. Go is the single source of truth for UI/core IPC.

const (
	ProtocolVersion = "0.1.0"
	Version         = ProtocolVersion
)

const (
	OpReady         = "ready"
	OpPing          = "ping"
	OpPong          = "pong"
	OpShutdown      = "shutdown"
	OpError         = "error"
	OpConnect       = "connect"
	OpDisconnect    = "disconnect"
	OpFramePush     = "frame.push"
	OpMessageIngest = "message.ingest"
	OpStatus        = "status"
	OpMessage       = "message"
	OpConfigSet     = "config.set"
	OpConfigGet     = "config.get"
	OpConfigValue   = "config.value" // response to config.get
	OpConfigOk      = "config.ok"    // ack after successful config.set (hub)
	OpCapabilities  = "capabilities"
	OpFocusUI       = "ui.focus" // hub → Pages: raise the existing window
	OpUICommand     = "ui.command"
	OpLoginState    = "login.state"
	OpRouteEnv      = "route.env"
	OpTick          = "tick"
	OpLedger        = "ledger"
	OpDanmuShow     = "danmu.show"
	OpMemoItem      = "memo.item"

	OpToolOvertimeSet = "tool.overtime.set"
	OpToolOvertimeCmd = "tool.overtime.cmd"
	OpToolOvertimeSim = "tool.overtime.sim_gift"
	OpToolDanmuSet    = "tool.danmu.set"
	OpToolMemoSet     = "tool.memo.set"
)

const (
	DefaultPipeName = `\\.\pipe\liveaio-core`
	DefaultTCPAddr  = "127.0.0.1:19877"
)
